### PR TITLE
[bug](regression) fix regression test test-table-version in cloud_p0

### PR DESCRIPTION
**Pull request by xy720**
## Proposed changes

Issue Number: close #xxx

Fix regression test test-table-version. 

introduce by:#32989

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xy720** [comment]
run buildall

**Comment by xy720** [comment]
run buildall

**Comment by xy720** [comment]
run buildall

**Comment by xy720** [comment]
run buildall

**Comment by xy720** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java [hunk 1]
```diff
@@ -1812,14 +1813,23 @@ public void dropPartition(Database db, OlapTable olapTable, DropPartitionClause
...
+            partition = olapTable.dropPartition(db.getId(), partitionName, clause.isForceDrop());
             if (!clause.isForceDrop() && partition != null) {
                 recycleTime = Env.getCurrentRecycleBin().getRecycleTimeById(partition.getId());
             }
         }
-        long version = olapTable.getNextVersion();
-        long versionTime = System.currentTimeMillis();
-        olapTable.updateVisibleVersionAndTime(version, versionTime);
+
+        long version = olapTable.getVisibleVersion();
```
[the remark this answers is not in the document]
Comment (xy720) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java:1822]: Users must obtain write lock of OlapTable before calling the dropPartition method.

[the remark this answers is not in the document]
Comment (xy720) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java:1822]: I found that the commit transaction holds a commit lock for the table. May be I need to hold a commit lock here?

**Review by gavinchou — APPROVED** [review]

**Pull request merged**